src/components/EResourceSections/Agreements.js

```diff
@@ -34,15 +38,17 @@ export default class Agreements extends React.Component {
   };
 
   renderEntitlementAgreements = () => {
-    const { entitlements = [] } = this.props.data;
-    const { headline, isEmptyMessage, visibleColumns } = this.props;
+    const { entitlements = [], entitlementsCount } = this.props.data;
+    const { handlers: { onNeedMoreEntitlements }, headline, isEmptyMessage, visibleColumns } = this.props;
 
     return (
       <EntitlementAgreementsList
         entitlements={entitlements}
         headline={headline}
         id="pci-agreements-list"
         isEmptyMessage={isEmptyMessage}
+        onNeedMoreData={onNeedMoreEntitlements}
```

Comment:
Can you use a different prop name here instead of `onNeedMoreData` as thats an MCL specific prop